Automattic/jetpack
# Replace offer reset plans checks by legacy plans checks

#### Changes proposed in this Pull Request:
In #17243, we added new colours to the `Banner` component to match the design of the new Jetpack plans introduced by the offer reset. Some variables and functions refer explicitly to the offer reset, which is bad naming since offer reset could potentially happen again multiple times in the future. This PR fixes that.

#### Jetpack product discussion
Issue brought up [here](https://github.com/Automattic/jetpack/pull/17243#pullrequestreview-495755548).

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
This PR requires to update the code locally to be tested, since all types of banners are not instanciated in the plugin.
* Download the PR locally and run the plugin
* Find an upsell banner, for instance the support upsell banner in _Jetpack > Dashboard > At a Glance > We're here to help_ (see captures below)
* Update the product/plan in `JETPACK_FEATURE_PRODUCT_UPSELL_MAP` in `_inc/client/lib/plans/constants.js` to test each of the 4 cases shown below. Keeping the same example mentioned above, update the value of `[FEATURE_PRIORITY_SUPPORT_JETPACK]` to `PLAN_JETPACK_PERSONAL`, `PLAN_JETPACK_SEARCH`, and `PLAN_JETPACK_COMPLETE` to test the legacy plan, product, and plan respectively.

**Legagy plan**
<img width="414" alt="Screen Shot 2020-09-30 at 11 10 07 AM" src="https://user-images.githubusercontent.com/1620183/94706850-b7074880-0310-11eb-8de9-b64fbf6532ae.png">

**Product**
<img width="422" alt="Screen Shot 2020-09-30 at 11 06 36 AM" src="https://user-images.githubusercontent.com/1620183/94706838-b2db2b00-0310-11eb-8979-a647b3797a0a.png">

**Plan**
<img width="420" alt="Screen Shot 2020-09-30 at 11 05 39 AM" src="https://user-images.githubusercontent.com/1620183/94706813-ace54a00-0310-11eb-8ee4-301eef5dec0b.png">

**Bundle**
<img width="413" alt="Screen Shot 2020-09-30 at 11 04 45 AM" src="https://user-images.githubusercontent.com/1620183/94706796-a656d280-0310-11eb-87bf-27e2759638f9.png">


#### Proposed changelog entry for your changes:
* Refactor variables and functions names for clarity
